honeybee_schema/energy/ventcool.py

```diff
@@ -93,13 +95,154 @@ class VentilationOpening(NoExtraBaseModel):
+        'two-way flow may occur due to stack effect.'
+    )
+
+
+class AFNCrack(NoExtraBaseModel):
+    """Properties for airflow through a crack."""
+
+    type: constr(regex='^AFNCrack$') = 'AFNCrack'
+
+    air_mass_flow_coefficient_reference: float = Field(
+        ...,
+        gt=0,
+        description='The air mass flow coefficient in kg/s at the conditions defined in '
+        'the AFNReferenceCrack condition, defined at a 1 Pa difference across this '
+        'crack.'
+    )
+
+    air_mass_flow_exponent: float = Field(
+        default=0.65,
+        ge=0.5,
+        le=1,
+        description='The air mass flow exponent for the surface crack. Default: 0.65.'
+    )
+
+    crack_factor: float = Field(
+        default=1,
+        gt=0,
+        le=1,
+        description='A number indicating multiplier for air mass flow through a crack.'
+        'Default: 1.'
+    )
+
+
+class AFNControlType(str, Enum):
+    multizone_with_distribution = 'MultiZoneWithDistribution'
+    multizone_without_distribution = 'MultiZoneWithoutDistribution'
+    multizone_with_distribution_only_during_fan_operation = \
+        'MultiZoneWithDistributionOnlyDuringFanOperation'
+
+
+class AFNBuildingType(str, Enum):
+    lowrise = 'LowRise'
+    highrise = 'HighRise'
+
+
+class AFNSimulationControl(NoExtraBaseModel):
+    """The global parameters used in the Airflow Network simulation."""
+
+    type: constr(regex='^AFNSimulationControl$') = 'AFNSimulationControl'
+
+    afn_control_type: AFNControlType = Field(
+        default=AFNControlType.multizone_without_distribution,
+        description='Text indicating type of control for an Airflow Network simulation. '
+        'Choices are: MultiZoneWithDistribution, MultiZoneWithoutDistribution, '
+        'or MultiZoneWithDistributionOnlyDuringFanOperation.'
+    )
+
+    building_type: AFNBuildingType = Field(
+        default=AFNBuildingType.lowrise,
+        description='Text indicating relationship between building footprint and '
+        'height used to calculate the wind pressure coefficients for exterior surfaces.'
+        'Choices are: LowRise and HighRise. LowRise corresponds to rectangular building '
+        'whose height is less then three times the width and length of the footprint. '
+        'HighRise corresponds to rectangular building whose height is more than three '
+        'times the width and length of the footprint. Default: LowRise.'
+    )
+
+    long_axis_angle: float = Field(
+        default=0,
+        ge=0,
+        le=180,
+        description='The clockwise rotation in degrees from true North of the long axis '
+        'of the building. Default: 0.'
+    )
+
+    aspect_ratio: float = Field(
+        default=1,
+        gt=0,
+        le=1,
+        description='Aspect ratio of a rectangular footprint, defined as the ratio of '
+        'length of the short axis divided by the length of the long axis. Default: 1.'
+    )
```

Comment:
This whole strategy of approximating the building as a rectangular footprint really seems less than ideal. Is there no way to pass the effective angles of individual window openings to the AFN such that the flow coefficients are computed from the relationship between the hourly EPW wind direction and the orientation of the opening.

This is what we are doing for the simple ventilation objects and you can see that there's an [effective angle property](https://bigladdersoftware.com/epx/docs/9-3/input-output-reference/group-airflow.html#field-effective-angle) that we use to pass the orientation of the window.

Reply:
@chriswmackey, let me get back to you on this. My understanding right now is based on the EP engineering reference reccomendations here: https://bigladdersoftware.com/epx/docs/9-0/engineering-reference/airflownetwork-model.html#wind-pressure-calculations, which is in fact also using the wind direction and orientation of the opening to derive the wind pressure coefficient - but they are including an additional parameter of the ratio of width of wall to with of adjacent wall, which is where the rectangular footprint constraint is coming from.

Reply:
Ok. I think I understand now. So the AFN will still compute the Cp for a given opening using its normal and the hourly wind direction. However, in order to do this, it has to first make a global assumption about the way air flows around the whole building on a macro-scale, which involves a simplified assumption of the building as an extruded rectangle. If my understanding here is correct, then I agree with your defaults and I think the aspect ratio is something that we can compute with ladybug_geometry whenever someone is exporting a model from Grasshopper with the intention of using the AFN. For this automated export, I think we might always use the same axis oriented to the world XY plane for simplicity but you are right to expose the axis orientation as an option on the schema in case advanced users who know their building geometry well want to improve the accuracy by better-orienting it to their geometry.